test/performance/search/index_construction_benchmark.cpp

```diff
@@ -0,0 +1,226 @@
+
+    seqan2_rng_t make_sequence_seqan2()
+    {
+        seqan2_rng_t result;
+
+        if constexpr(seqan3::dimension_v<rng_t> == 1)
+        {
+            seqan::resize(result, length);
+            std::transform(seqan3.begin(), seqan3.end(), seqan::begin(result), seqan3::to_char);
+        }
+        else
+        {
+            seqan::resize(result, collection_size);
+            for (size_t i = 0; i < collection_size; ++i)
+            {
+                seqan::resize(result[i], length);
+                std::transform(seqan3[i].begin(), seqan3[i].end(), seqan::begin(result[i]), seqan3::to_char);
+            }
+        }
+
+        return result;
+    }
+#endif // SEQAN3_HAS_SEQAN2
+
+public:
+    rng_t seqan3{make_sequence_seqan3()};
+
+    sequence_generator() = default;
+    sequence_generator(sequence_generator const &) = default;
+    sequence_generator(sequence_generator &&) = default;
+    sequence_generator & operator=(sequence_generator const &) = default;
+    sequence_generator & operator=(sequence_generator &&) = default;
+    ~sequence_generator() = default;
+    sequence_generator(size_t length_, size_t collection_size_) : length{length_}, collection_size(collection_size_) {};
+
+#if SEQAN3_HAS_SEQAN2
+    seqan2_rng_t seqan2{make_sequence_seqan2()};
+    using index_t = std::conditional_t<std::same_as<index_tag_t, fm_index_seqan2>,
+                                       fm_index_seqan2_t,
+                                       std::conditional_t<std::same_as<index_tag_t, bi_fm_index_seqan2>,
+                                                          bi_fm_index_seqan2_t,
+                                                          std::conditional_t<std::same_as<index_tag_t, fm_index_seqan3>,
+                                                                             fm_index_seqan3_t,
+                                                                             bi_fm_index_seqan3_t>>>;
+#else
+    using index_t = std::conditional_t<std::same_as<index_tag_t, fm_index_seqan3>,
+                                       fm_index_seqan3_t,
+                                       bi_fm_index_seqan3_t>;
+#endif
+};
+
+template <typename index_tag_t, typename rng_t>
+void index_benchmark(benchmark::State & state)
+{
+    sequence_generator<index_tag_t, rng_t> generator(state.range(0), state.range(1));
+    for (auto _ : state)
+    {
+#if SEQAN3_HAS_SEQAN2
+        if constexpr (std::same_as<index_tag_t, fm_index_seqan2> || std::same_as<index_tag_t, bi_fm_index_seqan2>)
+        {
+            typename sequence_generator<index_tag_t, rng_t>::index_t index{generator.seqan2};
+            seqan::indexCreate(index, seqan::FibreSALF());
+        }
+        else
+#endif // SEQAN3_HAS_SEQAN2
+            typename sequence_generator<index_tag_t, rng_t>::index_t index{generator.seqan3};
+    }
+}
+
+BENCHMARK_TEMPLATE(index_benchmark, fm_index_seqan3, std::vector<seqan3::dna4>)->Apply(arguments);
```

Comment:
You are generating the same sequences for every new benchmark right? This is quite inefficient for these many benchmarks :/ make we could store them somehow?
Sorry that this will require some more refactoring but we don't want the benchmark to take too much time.

Reply:
The sequence generation isn't acutally taking long, but I can refactor the tests.

The whole point of the class was to use the same sequences for seqan2 and seqan3, but I can just generate them individually and store them.

In this case, I would also block this PR by #920 such that I have the numeric sequence generator for generating char-sequences for seqan3 (numeric-sequence -> to_char?), because I think this is easier then generating a char sequence and then filtering out the ranks that are too high.

Reply:
I could actually just store a long enough sequence, and then do `| seqan3::views::drop(length)` on it. the view should produce overhead, but then again I'm not entirely sure ... but it would be a quite elegant solution?

Reply:
> In this case, I would also block this PR by #920 such that I have the numeric sequence generator for generating char-sequences for seqan3 (numeric-sequence -> to_char?), because I think this is easier than generating a char sequence and then filtering out the ranks that are too high.

:+1: 

> I could just store a long enough sequence, and then do | seqan3::views::take(length) on it. the view should produce overhead, but then again I'm not entirely sure ... but it would be a quite elegant solution?

Indeed this would be elegant :) And I think it would be comparable if you use 
`seqan3::views::take(length) | seqan3::views::to<std::vector>` ? This, of course, involves copying but this should not be a major performance bottleneck.. Or you could use vies::take for seqan3 and `Infix<>` for seqan2?

Reply:
Yes, something like this. I'll try it out and then we can talk about it on monday 👍